std/io.zig

```diff
@@ -224,6 +233,15 @@ pub const File = struct {
                     };
                 }
             },
+            Os.windows => {
+                if (system.SetFilePointerEx(self.handle, @bitCast(isize, pos), null, system.FILE_BEGIN) == 0) {
```

Comment:
bitcasting `pos` to `isize` could produce an invalid value if pos has a 1 for the highest-order bit. We should handle this edge case correctly. perhaps `std.math.cast` could be of use.